### ISSUE TITLE
The loss value of softDTW is inf when choosing bandwidth values

### ISSUE THREAD

**Issue by educationunion** [Maghoumi/pytorch-softdtw-cuda#18]
Hello @Maghoumi ,

I have tried the example codes you provided as follows, and I noticed that when choosing bandwidth = 1 or 2, the loss values are all inf. Could you please help me solve this issue? much appreciated.

Sample codes:

from soft_dtw_cuda import SoftDTW
# Create the sequences
batch_size, len_x, len_y, dims = 8, 15, 12, 5
x = torch.rand((batch_size, len_x, dims), requires_grad=True)
y = torch.rand((batch_size, len_y, dims))

-----------------------------------------
>>> sdtw = SoftDTW(use_cuda=False, gamma=0.1, bandwidth=2)
>>> loss = sdtw(x, y)
>>> loss
tensor([inf, inf, inf, inf, inf, inf, inf, inf], grad_fn=<_SoftDTWBackward>)
>>> sdtw = SoftDTW(use_cuda=False, gamma=0.1, bandwidth=1)
>>> loss = sdtw(x, y)
>>> loss
tensor([inf, inf, inf, inf, inf, inf, inf, inf], grad_fn=<_SoftDTWBackward>)

**Comment by Maghoumi** [comment]
Thanks for posting and example @educationunion! This issue was previously reported in #8 but I didn't have a good minimal example to try. I had a quick look, and I think this is caused by the condition check [around here](https://github.com/Maghoumi/pytorch-softdtw-cuda/blob/master/soft_dtw_cuda.py#L196) that fails to work as expected when the length of the two sequences being compared is different. This is perhaps the result of adding support for such sequences in ([this commit](https://github.com/Maghoumi/pytorch-softdtw-cuda/commit/7720c7d5552ae6698c75b0b7de832759dbe1dcd8)).

I think the condition has take the length of each sequence into account before skipping the calculation.
